src/Angor/Client/Services/FounderKeyService.cs

```diff
@@ -0,0 +1,80 @@
+using Angor.Shared.Models;       
+using Angor.Shared.Services;     
+using Nostr.Client.Messages;     
+using Angor.Client.Storage;
+
+public class FounderKeyService : IFounderKeyService
+{
+    private readonly IWalletStorage _walletStorage;
+    private readonly IRelayService _relayService;
+    private readonly ISerializer _serializer;
+    
+
+
+
+    public FounderKeyService(
+        IWalletStorage walletStorage,
+        IIndexerService indexerService,
+        IRelayService relayService,
+        IClientStorage storage,
+        ISerializer serializer)
+    {
+        _walletStorage = walletStorage;
+        _relayService = relayService;
+        _serializer = serializer;
+    }
+
+    public async Task<bool> IsFounderKeyInUseAsync(string founderKey)
+    {
+        var keys = _walletStorage.GetFounderKeys();
+        var inUseKeys = new HashSet<string>();
+
+        await _relayService.RequestProjectCreateEventsByPubKeyAsync(
+            keys.Keys.Select(k => k.NostrPubKey).ToArray(),
+            eventMessage =>
+            {
+                if (eventMessage.Kind == NostrKind.ApplicationSpecificData)
+                {
+                    var projectInfo = _serializer.Deserialize<ProjectInfo>(eventMessage.Content);
+                    inUseKeys.Add(projectInfo.FounderKey);
```

Comment:
this will fail if you get more then one result (multiple relays?)

Reply:
wait unless HashSet allows duplicates? but I think it does not or am I wrong?

Reply:
When you add an item to a HashSet that already contains that item, it ignores it.
So it doesn't look like a problem if both relays return the same value. 

( ran this thing to check,

```
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        HashSet<string> hashSet = new HashSet<string>();

        hashSet.Add("key1");
        hashSet.Add("key2");
        hashSet.Add("key1"); //duplicate
        hashSet.Add("key3");
        hashSet.Add("key2"); //dup

        foreach (var key in hashSet)
        {
            Console.WriteLine(key);
        }
    }
}
```
)